presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PickTableLayout.java

```diff
@@ -295,7 +295,7 @@ private static PlanNode planTableScan(
                         .map(node.getAssignments()::get)
                         .collect(toImmutableSet())));
 
-        if (layouts.isEmpty()) {
+        if (layouts.isEmpty() || layouts.stream().allMatch(layout -> newDomain.intersect(layout.getLayout().getPredicate()).isNone())) {
```

Comment:
is this not handled already? See line 308.

Reply:
It is indeed